docs/packaging/versioning.md

```diff
@@ -0,0 +1,161 @@
+| nightly releases     | `X.Y.ZaYYYYMMDD`  | `7.10.0a20251124`                                                                                                                                                   |
+| dev builds/releases  | `X.Y.Z.dev0+NNNN` | `7.10.0.dev0+efed3c3b10a5cce8578f58f8eb288582c26d18c4`<br>(for commit [`efed3c3`](https://github.com/ROCm/TheRock/commit/efed3c3b10a5cce8578f58f8eb288582c26d18c4)) |
+
+Each distribution channel (and GPU family within that channel) is currently
+hosted on a separate release index that can be selected with e.g.
+`pip install --index-url=https://rocm.nightlies.amd.com/v2/gfx94X-dcgpu/ rocm`.
+See [RELEASES.md - Installing releases using pip](../../RELEASES.md#installing-releases-using-pip)
+for details.
+
+### External Python package versions
+
+When we build external projects like
+[PyTorch](https://github.com/pytorch/pytorch) we sometimes extend the base
+package version with our own
+[local version identifier](https://packaging.python.org/en/latest/specifications/version-specifiers/#local-version-identifiers).
+
+For example, for torch version `2.9.0` built with ROCm version `7.9.0` we
+generate a composite torch version `2.9.0+rocm7.9.0`.
+
+<!-- TODO: Document [stable, rc, and dev] local version identifiers. -->
+
+<!-- See https://github.com/ROCm/TheRock/pull/2392 which may change for dev -->
+
+#### PyTorch versions
+
+PyTorch packages versions are handled via scripts:
+
+- [`build_tools/github_actions/determine_version.py`](/build_tools/github_actions/determine_version.py) (this generates e.g. `--version-suffix +rocm7.10.0`)
+  - [`build_tools/github_actions/tests/determine_version_test.py`](/build_tools/github_actions/tests/determine_version_test.py)
+- [`external-builds/pytorch/build_prod_wheels.py`](/external-builds/pytorch/build_prod_wheels.py) (this appends the version suffix to each build version)
+- [`build_tools/github_actions/write_torch_versions.py`](/build_tools/github_actions/write_torch_versions.py) (this finds the versions in built packages)
+
+The scripts produce these versions for each distribution channel:
+
+| Package name        | Example release version                                                           | Example nightly version        |
+| ------------------- | --------------------------------------------------------------------------------- | ------------------------------ |
+| torch               | `2.7.1+rocm7.9.0rc1`<br>_(future releases will drop the 'rc' suffix)_             | `2.10.0a0+rocm7.10.0a20251024` |
+| torchaudio          | `2.7.1a0+rocm7.9.0rc1`<br>_(future releases will drop the 'a' and 'rc' suffixes)_ | `2.10.0a0+rocm7.10.0a20251024` |
+| torchvision         | `0.22.1+rocm7.9.0rc1`<br>_(future releases will drop the 'rc' suffix)_            | `0.24.0+rocm7.11.0a20251124`   |
+| pytorch-triton-rocm | `3.3.1+rocm7.9.0rc1`<br>_(future releases will drop the 'rc' suffix)_             | `3.5.1+rocm7.11.0a20251124`    |
```

Comment:
Future _stable_ releases will drop the rc suffix, but assuming this is a release candidate, we wouldn't drop it. Maybe just drop the _(future releases will drop the 'rc' suffix)_ comment?